proc/breakpoints.go

```diff
@@ -110,14 +115,18 @@ func (dbp *Process) writeSoftwareBreakpoint(thread *Thread, addr uint64) error {
 }
 
 func (bp *Breakpoint) checkCondition(thread *Thread) bool {
-	if bp.Cond < 0 {
+	if bp.Cond == nil {
 		return true
 	}
-	g, err := thread.GetG()
+	scope, err := thread.Scope()
 	if err != nil {
-		return false
+		return true
```

Comment:
I don't think we really want to return `true` here. I understand that it would stop the debugger upon an error (essentially) but it could likely be a false positive, and yet there's no indication of that. We should either log/print a warning or modify the method signature to return an error and let the caller deal with it.